packages/file-explorer/src/FileListItemEditor.test.tsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import { act, render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {
+  FileListItemEditor,
+  FileListItemEditorProps,
+} from './FileListItemEditor';
+import { FileStorageItem } from './FileStorage';
+
+function makeItem(name = 'DEFAULT_NAME'): FileStorageItem {
+  return {
+    id: `id-${name}`,
+    type: 'file',
+    filename: name,
+    basename: name,
+  };
+}
+
+function makeWrapper({
+  item = makeItem(),
+  onCancel = jest.fn(),
+  onSubmit = jest.fn(),
+  validate,
+}: Partial<FileListItemEditorProps> = {}) {
+  return render(
+    <FileListItemEditor
+      item={item}
+      onCancel={onCancel}
+      onSubmit={onSubmit}
+      validate={validate}
+    />
+  );
+}
+
+describe('FileListItemEditor', () => {
+  it('mounts and unmounts without failing', () => {
+    const { unmount } = makeWrapper();
+    unmount();
+  });
+
+  it('validates on mount', () => {
+    const validate = jest.fn(() => Promise.resolve());
+    const itemName = 'test';
+    const { unmount } = makeWrapper({ item: makeItem(itemName), validate });
+    unmount();
+    expect(validate).toBeCalledWith(itemName);
+  });
+
+  it('validates on input changes', () => {
+    const validate = jest.fn(() => Promise.resolve());
+    const itemName = 'test';
+    const { unmount } = makeWrapper({ item: makeItem(itemName), validate });
+    const input: HTMLInputElement = screen.getByRole('textbox');
+    userEvent.type(input, '123');
+    unmount();
+    expect(validate).toBeCalledWith(`${itemName}1`);
+    expect(validate).toBeCalledWith(`${itemName}12`);
+    expect(validate).toBeCalledWith(`${itemName}123`);
+  });
+
+  it('validates and submits on enter', async () => {
+    const itemName = 'test';
+    const validate = jest.fn(() => Promise.resolve());
+    const onSubmit = jest.fn();
+    const onCancel = jest.fn();
+    const { unmount } = makeWrapper({
+      item: makeItem(itemName),
+      onCancel,
+      onSubmit,
+      validate,
+    });
+    const input: HTMLInputElement = screen.getByRole('textbox');
+    userEvent.type(input, '123{enter}');
+    // Wait to resolve the validation promise
+    await act(async () => {
+      await Promise.resolve();
+    });
```

Comment:
Could put this in a convenience function or in `TestUtils.flushPromises()`

Reply:
Updated